### ISSUE TITLE
Plugin crash during gradle sync

### ISSUE THREAD

**Issue by neworld** [Kotlin/kotlinx.serialization#97]
```
Found class org.jetbrains.kotlin.idea.facet.KotlinFacetConfiguration, but interface was expected
java.lang.IncompatibleClassChangeError: Found class org.jetbrains.kotlin.idea.facet.KotlinFacetConfiguration, but interface was expected
	at org.jetbrains.kotlinx.serialization.idea.KotlinSerializationImportHandler.modifyCompilerArguments(KotlinSerializationImportHandler.kt:56)
	at org.jetbrains.kotlinx.serialization.idea.KotlinSerializationGradleImportHandler.importBySourceSet(KotlinSerializationImportHandler.kt:33)
	at org.jetbrains.kotlin.idea.configuration.KotlinGradleSourceSetDataService.postProcess(KotlinGradleSourceSetDataService.kt:80)
	at com.intellij.openapi.externalSystem.service.project.manage.ProjectDataManagerImpl.lambda$doImportData$1(ProjectDataManagerImpl.java:296)
	at com.intellij.openapi.externalSystem.service.project.manage.ProjectDataManagerImpl.importData(ProjectDataManagerImpl.java:150)
	at com.intellij.openapi.externalSystem.service.project.manage.ProjectDataManagerImpl.importData(ProjectDataManagerImpl.java:221)
	at com.intellij.openapi.externalSystem.service.project.manage.ProjectDataManagerImpl.importData(ProjectDataManagerImpl.java:228)
	at com.intellij.openapi.externalSystem.util.ExternalSystemUtil$MyMultiExternalProjectRefreshCallback.onSuccess(ExternalSystemUtil.java:1149)
	at com.intellij.openapi.externalSystem.util.ExternalSystemUtil$3.execute(ExternalSystemUtil.java:555)
	at com.intellij.openapi.externalSystem.util.ExternalSystemUtil$5.run(ExternalSystemUtil.java:612)
	at com.intellij.openapi.progress.impl.CoreProgressManager$TaskRunnable.run(CoreProgressManager.java:713)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$1(CoreProgressManager.java:157)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:543)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:488)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:94)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:144)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl$2.run(ProgressManagerImpl.java:165)
	at com.intellij.openapi.application.impl.ApplicationImpl$1.run(ApplicationImpl.java:315)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

It happens in the end of the sync and prevents refresh dependencies. 

Kotlin: 1.2.30
Plugin: 0.4.2
IntelliJ: 2017.3.4 CE

**Comment by sandwwraith** [comment]
Seems the latest master branch is compatible with `1.2.30` CLI compiler, but not with `1.2.30` IDEA compiler. Current workaround is to use `1.2.40-dev-XXX` plugins for IDEA

**Comment by sandwwraith** [comment]
Better to try out recently published 1.2.30-compatible IDEA plugin
